js/playlist.js

```diff
@@ -94,30 +101,36 @@ function(ytData,ytIframe,template,$) {
     }
 
     var saveDuration = function() {
-        localStorage.setItem('total_duration', JSON.stringify(total_duration));
+        durations[curr_id]=total_duration;
+        localStorage.setItem('duration', JSON.stringify(durations));
     }
 
     var savePlaylist = function() {
         localStorage.setItem('playlist', JSON.stringify(playlist));
         localStorage.setItem('new_playlist_saved', JSON.stringify(true));
     }
 
+    var saveRepeat = function(mode){
```

Comment:
If you change to having a common repeat option for every playlist, let it stay in `player.js`, because repeat option is a player specific property not a playlist specific property.